gslib/command.py

```diff
@@ -1929,6 +1953,61 @@ def _ApplyThreads(self, thread_count, process_count, recursive_apply_level,
         # block the semaphore on a WorkerThread completion.
         worker_semaphore.release()
 
+  def _GetGcloudStorageArgs(self, sub_opts, gsutil_args, gcloud_storage_map):
```

Comment:
I'm thinking this should also be in a separate file, but I see it relies on instance variables. Maybe a subclass solution is possible or a util that gets `self` as an argument (except renamed)?

Reply:
Subclass is not an option because this class is the parent of all the other commands. If we decide to subclass it, we will have to update the parent for all the commands, which seems unnecessary just to handle a new method.

I personally don't like passing in an entire object (self) to a util either. I feel this method belongs in this class. Like I said, there are other potential candidates if we want to refactor the code here, which I agree has grown a lot. But given the priority of the team, I don't think refactoring is going to happen anytime soon, so I would prefer leaving it like this.

Reply:
On a second thought, I think we can pass over the entire command instance from command runner. 
What I typically like to avoid  is having a circular reference (a method calls a util with it's own instance and then the util calls a method on that instance).  But looking at the code in CommandRunner, it creates the command instance and calls the RunCommand on it. We will be eventually calling something like `RunGcloudStorage` and for that, we can probably pass over the entire command instance to the util which also avoids the circular reference issue mentioned above and at the same time move this logic to the util. I will go ahead and make that change.

Reply:
Or you could only pass the variables of from the CommandRunner intance you need

Reply:
I think passing `Command` instance is better because there a lot of things that we need from that instance. Plus we also call the `ParseSubOpts` method on the instance.

Reply:
I should've checked this before I moved the method. I revisited my design doc and realized why I had decided to keep GetGcloudStorageArgs under `Command` class. We want the ability to override this generic behavior if a command is too complicated to be translated using this generic method. So having this under `Command` class makes more sense to me. An example would be signurl where a service account key file is a positional argument in gsutil but it's going to be a required flag in gcloud storage. Reverted to keeping this method under `Command`.

Reply:
What about moving the gcloud logic to a mixin class?

Reply:
Good idea. Mixin should work.

Reply:
Done.